ppoprf/src/ppoprf.rs

```diff
@@ -87,6 +148,143 @@ impl ProofDLEQ {
+    let t2 = (self.s * RISTRETTO_BASEPOINT_POINT) + (self.c * public_value);
+    //t3 = ((s * M) + (c * Z))
+    let t3 = (self.s * m) + (self.c * z);
+
+    /*challengeTranscript =
+    I2OSP(len(Bm), 2) || Bm ||
+    I2OSP(len(a0), 2) || a0 ||
+    I2OSP(len(a1), 2) || a1 ||
+    I2OSP(len(a2), 2) || a2 ||
+    I2OSP(len(a3), 2) || a3 ||
+    "Challenge"*/
+    let mut challenge_transcript = Vec::new();
+    challenge_transcript.extend_from_slice(ProofDLEQ::i2osp(
+      &public_value.compress().as_bytes().len().to_be_bytes(),
+      2,
+    ));
+    challenge_transcript.extend_from_slice(public_value.compress().as_bytes()); //Bm
+    challenge_transcript.extend_from_slice(ProofDLEQ::i2osp(
+      &m.compress().as_bytes().len().to_be_bytes(),
+      2,
+    ));
+    challenge_transcript.extend_from_slice(m.compress().as_bytes()); //a0 = m
+    challenge_transcript.extend_from_slice(ProofDLEQ::i2osp(
+      &z.compress().as_bytes().len().to_be_bytes(),
+      2,
+    ));
+    challenge_transcript.extend_from_slice(z.compress().as_bytes()); //a1 = z
+    challenge_transcript.extend_from_slice(ProofDLEQ::i2osp(
+      &t2.compress().as_bytes().len().to_be_bytes(),
+      2,
+    ));
+    challenge_transcript.extend_from_slice(t2.compress().as_bytes()); //a2 = t2
+    challenge_transcript.extend_from_slice(ProofDLEQ::i2osp(
+      &t3.compress().as_bytes().len().to_be_bytes(),
+      2,
+    ));
+    challenge_transcript.extend_from_slice(t3.compress().as_bytes()); //a3 = t3
+
+    //c = G.HashToScalar(challengeTranscript)
+    let mut out = [0u8; 64];
+    strobe_hash(&challenge_transcript, "Challenge", &mut out);
+    let expected_c = RistrettoScalar::from_bytes_mod_order_wide(&out);
+
+    //verified = (expectedC == c)
+    return expected_c == self.c;
+  }
+
+  fn compute_composites(
+    key: Option<RistrettoScalar>,
+    b: &RistrettoPoint,
+    c: &[RistrettoPoint],
+    d: &[RistrettoPoint],
+  ) -> (RistrettoPoint, RistrettoPoint) {
+    if c.len() != d.len() {
+      panic!("C and D have a different number of elements!");
+    }
+
+    // We use the Partially-punctureable Oblivious Pseudo-Random Function
+    let context_string =
+      format!("{}{}{}{}", "PPOPRFv1-", 0x03, "-", "ristretto255-strobe");
+    //seedDST = "Seed-" || contextString
+    let seed_dst = format!("{}{}", "Seed-", context_string);
+
+    // seedTranscript = I2OSP(len(Bm), 2) || Bm || I2OSP(len(seedDST), 2) || seedDST
+    let mut seed_transcript = Vec::new();
+    seed_transcript.extend_from_slice(ProofDLEQ::i2osp(
+      &b.compress().as_bytes().len().to_be_bytes(),
+      2,
+    ));
+    seed_transcript.extend_from_slice(b.compress().as_bytes());
+    seed_transcript
+      .extend_from_slice(ProofDLEQ::i2osp(&seed_dst.len().to_be_bytes(), 2));
+    seed_transcript.extend_from_slice(seed_dst.as_bytes());
+
+    let mut seed = [0u8; 64];
+    strobe_hash(&seed_transcript, "Seed", &mut seed);
+
+    //M = G.Identity()
+    let mut m = RISTRETTO_BASEPOINT_POINT;
```

Comment:
The identity element is not the base_point. It should be: https://github.com/cfrg/draft-irtf-cfrg-voprf/blob/main/poc/ristretto_decaf.sage#L342